src/core/server/saved_objects/service/lib/repository.security_extension.test.ts

```diff
@@ -0,0 +1,1824 @@
+      expect(mockSecurityExt.enforceAuthorization).toHaveBeenCalledWith(
+        expect.objectContaining({
+          action: 'bulk_get',
+        })
+      );
+
+      const expectedTypesAndSpaces = new Map([
+        [objA.type, new Set([optionsNamespace, ...objA.namespaces])],
+        [objB.type, new Set([optionsNamespace, ...objB.namespaces])],
+      ]);
+
+      const { typesAndSpaces: actualTypesAndSpaces, typeMap: actualTypeMap } =
+        mockSecurityExt.enforceAuthorization.mock.calls[0][0];
+
+      expect(mapsAreEqual(actualTypesAndSpaces, expectedTypesAndSpaces)).toBeTruthy();
+      expect(actualTypeMap).toBe(authMap); // ToDo? reference comparison ok, object is frozen
+    });
+
+    test(`calls redactNamespaces with authorization map`, async () => {
+      setupCheckAuthorized(mockSecurityExt);
+      setupRedactPassthrough(mockSecurityExt);
+
+      const objects = [obj1, obj2];
+
+      await bulkGetSuccess(client, repository, registry, objects, { namespace });
+
+      expect(mockSecurityExt.checkAuthorization).toHaveBeenCalledTimes(1);
+      expect(mockSecurityExt.enforceAuthorization).toHaveBeenCalledTimes(1);
+
+      expect(mockSecurityExt.redactNamespaces).toHaveBeenCalledTimes(2);
+
+      objects.forEach((obj, i) => {
+        const { savedObject, typeMap } = mockSecurityExt.redactNamespaces.mock.calls[i][0];
+        expect(savedObject).toEqual(
+          expect.objectContaining({
+            type: obj.type,
+            id: obj.id,
+            namespaces: [namespace],
+          })
+        );
+        expect(typeMap).toBe(authMap);
+      });
+    });
+
+    test(`adds audit event per object when successful`, async () => {
+      setupCheckAuthorized(mockSecurityExt);
+      setupRedactPassthrough(mockSecurityExt);
+      setupEnforceSuccess(mockSecurityExt);
+
+      const objects = [obj1, obj2];
+      await bulkGetSuccess(client, repository, registry, objects, { namespace });
+
+      expect(mockSecurityExt.addAuditEvent).toHaveBeenCalledTimes(objects.length);
+      objects.forEach((obj) => {
+        expect(mockSecurityExt.addAuditEvent).toHaveBeenCalledWith({
+          action: AuditAction.GET,
+          savedObject: { type: obj.type, id: obj.id },
+        });
+      });
+    });
+
+    test(`adds audit event per object when not successful`, async () => {
+      setupCheckAuthorized(mockSecurityExt);
+      setupEnforceFailure(mockSecurityExt);
+
+      const objects = [obj1, obj2];
+      await expect(
+        bulkGetSuccess(client, repository, registry, objects, { namespace })
+      ).rejects.toThrow(enforceError);
+
+      expect(mockSecurityExt.addAuditEvent).toHaveBeenCalledTimes(objects.length);
+      objects.forEach((obj) => {
+        expect(mockSecurityExt.addAuditEvent).toHaveBeenCalledWith({
+          action: AuditAction.GET,
+          savedObject: { type: obj.type, id: obj.id },
+          error: enforceError,
+        });
+      });
+    });
+  });
+
+  describe('#bulkCreate', () => {
```

Comment:
Similar question from `create`: Do you think it's worthwhile to include tests for multi-namespace types, since they influence what we pass into the security extension?

Reply:
We do test the initial namespaces path. Do you mean adding a case where we mock an existing multi-space object with a different set of spaces than what is passed in for initial spaces? This would be applicable to create, but I think bulkCreate will just resort to using the namespace option/active space in that case.

Are there more cases outside of initial "namespaces"?

Reply:
It looks like this code block isn't exercised by the current test suite, which is for existing multi-namespace types, both with and without `initialNamespaces` provided:

https://github.com/jportner/kibana/blob/d8110c4b9e858d2a1996a43e13fe99158e21b486/src/core/server/saved_objects/service/lib/repository.ts#L675-L677